### PR TITLE
Implement dashboard page for Open Voice Challenge

### PR DESCRIPTION
**Pull request by catherine917**
@rileyjshaw  As a dashboard is a huge task, I created a pull request to update my work progress.  Please don't merge this branch to the master branch until all functions completed.

### CODE REVIEW

**Comment by rileyjshaw** [comment]
This is a fantastic start, thanks @catherine917! You rock.

File: web/src/services/api.ts [hunk 1]
```diff
@@ -280,13 +303,66 @@ export default class API {
       API_PATH +
         '/user_client/awards/seen' +
         (kind == 'notification' ? '?notification' : ''),
-      {
-        method: 'POST',
-      }
+      { method: 'POST' }
```
[the remark this answers is not in the document]
Comment (catherine917) [reply web/src/services/api.ts]: Yep, I use the format plugin named `Pretty Formatter ` in the vscode. Maybe we can use the same tool to formate the document.

File: web/src/services/api.ts [hunk 2]
```diff
@@ -280,13 +303,66 @@ export default class API {
...
+  }
+
+  // Challenge
+  fetchChallengePoints(
+    email?: string
+  ): Promise<{
+    user: number;
+    team: number;
+  }> {
+    return this.fetch(`${API_PATH}/challenge/points/${email}`);
```
[the remark this answers is not in the document]
Comment (catherine917) [reply web/src/services/api.ts]: Sure. I will align with @can-007

[the remark this answers is not in the document]
Comment (can-007) [reply web/src/services/api.ts]: Yeah, client_id is already be include, we will remove email.

For future-proof, the challenge id or challenge url_token?

[the remark this answers is not in the document]
Comment (rileyjshaw) [reply web/src/services/api.ts]: We'll want to use the `url_token`.

File: web/src/services/api.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (rileyjshaw) [reply web/src/services/api.ts]: Ah, got it! You can run our formatter with `yarn run prettier`.

[the remark this answers is not in the document]
Comment (rileyjshaw) [reply web/src/services/api.ts]: There is also a prettier extension for VS Code here: https://github.com/prettier/prettier-vscode

**Pull request merged**